src/api/forms/FormController.js

```diff
@@ -0,0 +1,72 @@
+import AutoCompleteField from './components/controls/AutoCompleteField.vue';
+import ClockDisplayFormatField from './components/controls/ClockDisplayFormatField.vue';
+import Datetime from './components/controls/Datetime.vue';
+import FileInput from './components/controls/FileInput.vue';
+import Locator from './components/controls/Locator.vue';
+import NumberField from './components/controls/NumberField.vue';
+import SelectField from './components/controls/SelectField.vue';
+import TextAreaField from './components/controls/TextAreaField.vue';
+import TextField from './components/controls/TextField.vue';
+
+export const DEFAULT_CONTROLS_MAP = {
```

Comment:
I think there's a misunderstanding here. This cannot be hard-coded. Think about how an outsider developer would contribute a new form control. An analogy here is views. We do not have a map of all of the possible views in Open MCT because a third party developer would never be able to define a new view.

Again, the original issue had an example of how we might do define the API for this, which is based on how the view API handles this -
``` JavaScript
openmct.forms.addControl('colorPicker', {
    show(element) {
        // A render function
    },
    destroy() {
        // A destroy function
    }
});
```

Reply:
hmm, I have implemented/tried this way, let me know if following example makes sense,

###  1. Adding new Control from any file (here I tried from notebook plugin.js)
```
import CustomControl from './CustomControl.vue';


.....


openmct.forms.addNewFormControl('customControl', {
            show: () => {
                return CustomControl;
            },
            destroy() {
                console.log('destroyed');
            }
        });
```

### 2. Use this `customControl` in form properties
```
{
      key: 'custom',
      name: 'Custom Control',
      control: 'customControl',
      cssClass: 'l-inline',
      required: false,
      property: [
          "configuration",
          "custom"
      ]
  }
```
### 3. Define CustomControl.vue  
(exactly same as  `src/api/forms/components/controls/TextField.vue`)
<img width="454" alt="Screen Shot 2021-09-17 at 3 45 33 PM" src="https://user-images.githubusercontent.com/1292612/133861194-3e4f97be-4dec-4f13-a0da-068a7adb6953.png">

Reply:
Almost. The problem is that the custom control cannot just map to a Vue Component. That forces third party developers to use Vue. In my example the `show(element)` function accepts a DOM element as an argument. In our case we would just mount a Vue component at this location (the exact same way our view API works). A third party developer could use that dom element to mount a react component, or a vanilla JS component, or whatever.